### ISSUE TITLE
Simplify `if x then True else y` to `x || y`

### ISSUE THREAD

**Issue by SiriusStarr** [jfmengels/elm-review-simplify#12]
Stumbled across it in my (old, shameful) code the other day, but the following simplifications could be added:

```elm
if foo then
    True
else
    bar
```

to

```elm
foo || bar
```

and similarly

```elm
if too then
    bar
else
    False
```

to

```elm
foo && bar
```

Should be easy to implement if you agree they should be, but don't want to put time into it if you disagree.  :smile:

**Comment by jfmengels** [comment]
Hmm, this is interesting. I like it, but I'd love some more external opinions (adding new kinds of simplifications to this existing rule can bother some people).

One thing though, I'm thinking that maybe the rule shouldn't do anything if there are comments anywhere inside the if, as the developer may have split it on purpose for readability purposes. What do you think?

**Comment by SiriusStarr** [comment]
I agree that not swallowing comments would probably be a good idea; if it's commented, presumably it's that way for readability and not developer oversight.  Though that would make it more of a challenge to implement, I would think?  Unless there's a clever way to check for comments in `elm-syntax`?  I assume you have to make a comment visitor to save all the locations and then pass that context to the expression visitor?

**Comment by jfmengels** [comment]
> I assume you have to make a comment visitor to save all the locations and then pass that context to the expression visitor?

Yes exactly 👌 in practice I think that keeping the starting line numbers for each comment would be enough.

**Comment by SiriusStarr** [comment]
> Yes exactly ok_hand in practice I think that keeping the starting line numbers for each comment would be enough.

Awesome, I actually need to do this for another rule, so I appreciate the input.  :rofl:

**Comment by jfmengels** [comment]
I'm also thinking that if you have an else if, in the second branch, this rule should not report anything

The example from the original comment:
```elm
if foo then
    True
else
    bar
--> turns into
foo || bar
```

With an else if
```elm
if foo then
    True
else if otherCondition then
    bar
else
    baz
--> would turn into
foo ||
    if otherCondition then
		bar
    else
	    baz
```
and I think this looks a bit odd.

I imagine the same thing to happen if in the second branch you have a let expression or case expression, because that would look quite odd and therefore not a likely improvement.

**Comment by SiriusStarr** [comment]
That seems reasonable to me.  It's only the base case that seems clearly made simpler by it.

**Comment by hingew** [comment]
Should this rule be part of the `Simply` rule?

**Comment by lue-bird** [comment]
We'd also need to check if one branch is a recursive call to not remove TCO by fixing to `||` or `&&`.
